policy-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/policy/kafka/KafkaContractTest.java

```diff
@@ -0,0 +1,359 @@
+import org.springframework.core.env.Environment;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.core.serializer.support.SerializationFailedException;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.support.TestPropertySourceUtils;
+import org.springframework.util.LinkedMultiValueMap;
+import org.testcontainers.containers.KafkaContainer;
+import scala.concurrent.duration.FiniteDuration;
+
+import uk.gov.gchq.palisade.contract.policy.ContractTestData;
+import uk.gov.gchq.palisade.service.policy.PolicyApplication;
+import uk.gov.gchq.palisade.service.policy.model.AuditErrorMessage;
+import uk.gov.gchq.palisade.service.policy.model.PolicyRequest;
+import uk.gov.gchq.palisade.service.policy.model.StreamMarker;
+import uk.gov.gchq.palisade.service.policy.model.Token;
+import uk.gov.gchq.palisade.service.policy.stream.ConsumerTopicConfiguration;
+import uk.gov.gchq.palisade.service.policy.stream.ProducerTopicConfiguration;
+import uk.gov.gchq.palisade.service.policy.stream.PropertiesConfigurer;
+import uk.gov.gchq.palisade.service.policy.stream.SerDesConfig;
+
+import java.io.IOException;
+import java.util.AbstractMap;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toMap;
+import static org.apache.kafka.clients.admin.AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+/**
+ * An external requirement of the service is to connect to a pair of kafka topics.
+ * The upstream "resource" topic is written to by the resource-service and read by this service.
+ * The downstream "rule" topic is written to by this service and read by the attribute-masking-service.
+ * Upon writing to the upstream topic, appropriate messages should be written to the downstream topic.
+ */
+@SpringBootTest(
+        classes = PolicyApplication.class,
+        webEnvironment = WebEnvironment.RANDOM_PORT,
+        properties = {"akka.discovery.config.services.kafka.from-config=false", "spring.cache.caffeine.spec=expireAfterWrite=1m, maximumSize=100"}
+)
+@Import({KafkaContractTest.KafkaInitializer.Config.class})
+@ContextConfiguration(initializers = {KafkaContractTest.KafkaInitializer.class})
+@ActiveProfiles({"caffeine", "akkatest", "pre-population"})
+class KafkaContractTest {
```

Comment:
Test runs for me locally, but there is a subtitle issue with it.  The test is dependant on settings on my computer.  That means that potentially each of us can get a different outcome.    In this case I don't think it matters, but shouldn't.   
In this case I see the message "To enable reuse of containers, you must set 'testcontainers.reuse.enable=true' in a file located at ....testcontainers.properties"

Reply:
this would be a performance upgrade but is not crucial to running the test